contracts/FullCheckpoint.sol

```diff
@@ -413,43 +422,39 @@ contract FullCheckpoint {
 
     /*
      * @param subnet block number.
-     * @return BlockLite struct defined above.
+     * @return HeaderInfo struct defined above.
      */
     function getHeaderByNumber(
-        int256 number
-    ) public view returns (BlockLite memory) {
-        if (committedBlocks[number] == 0) {
-            int256 blockNum = int256(
-                uint256(uint64(headerTree[latestBlock].mix >> 128))
-            );
-            if (number > blockNum) {
-                return BlockLite({hash: bytes32(0), number: 0});
-            }
-            int256 numGap = blockNum - number;
-            bytes32 currHash = latestBlock;
-            for (int256 i = 0; i < numGap; i++) {
-                currHash = headerTree[currHash].parentHash;
+        uint256 number
+    ) public view returns (HeaderInfo memory) {
+        return getHeader(heightTree[uint64(number)]);
+    }
+
+    /**
+     * @dev Get the finalized information of a block.
+     * @param blockHash  The hash of the block.
+     * @return finalizedNumber  The number of the block.
+     * @return isFinalized  Whether the block is finalized.
+     */
+    function getFinalizedInfo(
+        bytes32 blockHash
+    ) public view returns (int256 finalizedNumber, bool isFinalized) {
+        uint256 blockNumber = uint256(uint64(headerTree[blockHash].mix >> 128));
+        if (blockNumber != 0) {
+            for (uint256 i = 0; i < finalizedRanges.length; i++) {
+                if (
+                    blockNumber > finalizedRanges[i].lastFinalizedNumber &&
+                    blockNumber <= finalizedRanges[i].latestFinalizedNumber
+                ) {
+                    return (
+                        int256(finalizedRanges[i].latestFinalizedNumber),
+                        true
+                    );
+                }
```

Comment:
is there another condition to break the loop?
`if (...) break`
break can save gas

Reply:
here can not break , find this is required